File: src/main/java/link/infra/packwiz/installer/ui/InstallProgress.java
package link.infra.packwiz.installer.ui;

public class InstallProgress {
	public final String message;
	public final boolean hasProgress;
	public final int progress;
	public final int progressTotal;

	public InstallProgress(String message) {
		this.message = message;
		hasProgress = false;
		progress = 0;
		progressTotal = 0;
	}

	public InstallProgress(String message, int progress, int progressTotal) {
		this.message = message;
		hasProgress = true;
		this.progress = progress;
		this.progressTotal = progressTotal;
	}
}

File: src/main/kotlin/link/infra/packwiz/installer/ui/InstallProgress.kt
package link.infra.packwiz.installer.ui

data class InstallProgress(
		val message: String,
		val hasProgress: Boolean = false,
		val progress: Int = 0,
		val progressTotal: Int = 0
) {
	constructor(message: String, progress: Int, progressTotal: Int) : this(message, true, progress, progressTotal)

	constructor(message: String) : this(message, false)
}